packages/in_app_purchase/example/lib/main.dart

```diff
@@ -0,0 +1,372 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: avoid_print
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:in_app_purchase/in_app_purchase.dart';
+import 'package:in_app_purchase_tizen/billing_manager_wrappers.dart';
+import 'package:in_app_purchase_tizen/in_app_purchase_tizen.dart';
+
+void main() {
+  WidgetsFlutterBinding.ensureInitialized();
+
+  InAppPurchaseTizenPlatform.registerPlatform();
+  runApp(_MyApp());
+}
+
+// To try without auto-consume, change `true` to `false` here.
+const bool _kAutoConsume = true;
+
+/// The [_kConsumableType]'s value:
+/// "1": CONSUMABLE
+/// "2": NON-CONSUMABLE
+/// "3": LIMITED-PERIOD
+/// "4": SUBSCRIPTION
+const int _kConsumableType = 1;
+const String _kAppId = '3201504002021';
+const String _kCustumId = '810000047372';
+const String _kServerType = 'DEV';
+const String _kCountryCode = 'US';
+const String _kItemType = '2';
+const String _kPageSize = '20';
+const String _kPageNum = '1';
+// ignore: unused_element
+const String _kSecurityKey = '7db0YeFxHPtajwvVShBC5auI4iSJraKy8yhPZe13yvM=';
+
+/// The check value is used by the DPI service to verify Purchase List and Products List API requests.
+/// It is a Base64 hash generated by applying the HMAC SHA256 algorithm on a concatenated string of parameters using the DPI security key.
+///
+/// The [_kProductCheckValue] is generated with [_kAppId] and [_kCountryCode] and [_kSecurityKey];
+/// var hmacSha256 = Hmac(sha256, utf8.encode(_kSecurityKey));
+/// String _kProductCheckValue = base64Url.encode(hmacSha256.convert(utf8.encode(_kAppId + _kCountryCode)).bytes);
+String _kProductCheckValue = 'sbKzQJrFdeLkfXTQUdjjq4jt7ykx79QmJno9ZLam+SM=';
+
+/// The [_kPurchaseCheckValue] is generated with [_kAppId] and [_kCustumId] and [_kCountryCode] and [_kItemType] and [_kSecurityKey];
+/// var hmacSha256 = Hmac(sha256, utf8.encode(_kSecurityKey));
+/// String _kPurchaseCheckValue = base64Url.encode(hmacSha256.convert(utf8.encode(_kAppId + _kCustumId + _kCountryCode + _kItemType)).bytes);
+String _kPurchaseCheckValue = 'eAKHVR/FUfgEmDb3KmtgkozWAKyEb1OWXKdvkUV4km0=';
+
+// WARNING: Don't change order of `_kRequestParams`.
+List<String> _kRequestParams = <String>[
+  _kAppId,
+  _kCountryCode,
+  _kItemType,
```

Comment:
What is the item type? The parameter is not documented in the [API references](https://developer.samsung.com/smarttv/develop/api-references/samsung-product-api-references/billing-api.html#BillingManager-getProductsList).

Reply:
You're passing the value when retrieving the product list but not the purchase list. That's not the right place. Also I still doubt that you need to pass the value since

- The value is literally unused. The value is used to calculate `checkValue` but you already hardcoded the `checkValue`.
- The value is a constant (2), so the plugin already knows the value. There's no need to require the user to provide the value.

Reply:
Ok, I know, I will change it.

Reply:
fixed

Reply:
Set `itemType` as a constant value '2', hardcoded into requestPurchases.